std/haxe/Rest.hx

```diff
@@ -0,0 +1,58 @@
+package haxe;
+
+import haxe.iterators.RestKeyValueIterator;
+import haxe.iterators.RestIterator;
+
+@:coreType
+abstract Rest<T> {
```

Comment:
`to Iterable<T>` for static extension support might be nice?

Reply:
No, because it's not compatible with `Iterable`.
The only thing we can to is adding `@:to Iterable`

Reply:
Hm... but it has `iterator()`?

Reply:
But it only exists at compile time, while `Iterable` needs `iterator` to exist at run time.